File: services/llm-gateway/src/model_router.py
from __future__ import annotations

import os
from typing import Any


def route_model(task: str, scope: dict[str, Any]) -> dict[str, str]:
    return {
        "provider": "deepseek",
        "model": os.environ.get("DEEPSEEK_MODEL", "deepseek-v4-flash"),
        "task": task,
        "role": scope.get("role", "unknown"),
    }

File: services/llm-gateway/ts-src/modelRouter.ts
import { deepseekModel } from "./config.js";

export function routeModel(
  task: string,
  scope: Record<string, unknown>,
): Record<string, string | null> {
  return {
    provider: "deepseek",
    model: deepseekModel(),
    task: task,
    role: "role" in scope ? (scope["role"] as string | null) : "unknown",
  };
}
